android/src/toga_android/factory.py

```diff
@@ -1,86 +1,62 @@
 
+toga_android_factory_imports = {
+    "App": "toga_android.app",
+    "MainWindow": "toga_android.app",
+    "Command": "toga_android.command",
+    "Camera": "toga_android.hardware.camera",
+    "Font": "toga_android.fonts",
+    "Icon": "toga_android.icons",
+    "Image": "toga_android.images",
+    "Paths": "toga_android.paths",
+    "Box": "toga_android.widgets.box",
+    "Button": "toga_android.widgets.button",
+    "Canvas": "toga_android.widgets.canvas",
+    "DateInput": "toga_android.widgets.dateinput",
+    "DetailedList": "toga_android.widgets.detailedlist",
+    "Divider": "toga_android.widgets.divider",
+    "dialogs": "toga_android",
+    "ImageView": "toga_android.widgets.imageview",
+    "Label": "toga_android.widgets.label",
+    "Location": "toga_android.hardware.location",
+    "MapView": "toga_android.widgets.mapview",
+    "MultilineTextInput": "toga_android.widgets.multilinetextinput",
+    "NumberInput": "toga_android.widgets.numberinput",
+    "OptionContainer": "toga_android.widgets.optioncontainer",
+    "PasswordInput": "toga_android.widgets.passwordinput",
+    "ProgressBar": "toga_android.widgets.progressbar",
+    "ScrollContainer": "toga_android.widgets.scrollcontainer",
+    "Selection": "toga_android.widgets.selection",
+    "Slider": "toga_android.widgets.slider",
+    "Switch": "toga_android.widgets.switch",
+    "Table": "toga_android.widgets.table",
+    "TextInput": "toga_android.widgets.textinput",
+    "TimeInput": "toga_android.widgets.timeinput",
+    "WebView": "toga_android.widgets.webview",
+    "Window": "toga_android.window",
+}
+__all__ = list(toga_android_factory_imports.keys()) + ["not_implemented"]
 
-def not_implemented(feature):
-    NotImplementedWarning.warn("Android", feature)
 
+def __getattr__(name):
+    try:
+        module_name = toga_android_factory_imports[name]
 
-__all__ = [
-    "App",
-    "Command",
-    "MainWindow",
-    "not_implemented",
-    # Resources
-    "dialogs",
-    "Font",
-    "Icon",
-    "Image",
-    "Paths",
-    # Hardware
-    "Camera",
-    "Location",
-    # Widgets
-    # ActivityIndicator
-    "Box",
-    "Button",
-    "Canvas",
-    "DateInput",
-    "DetailedList",
-    "Divider",
-    "ImageView",
-    "Label",
-    "MapView",
-    "MultilineTextInput",
-    "NumberInput",
-    "OptionContainer",
-    "PasswordInput",
-    "ProgressBar",
-    "ScrollContainer",
-    # "SplitContainer",
-    "Selection",
-    "Slider",
-    "Switch",
-    "Table",
-    "TextInput",
-    "TimeInput",
-    # "Tree",
-    "WebView",
-    "Window",
-]
+        has_dot = module_name.find(".") != -1
+        if not has_dot:
+            module_name = f"toga_android.{name}"
+    except KeyError:
+        raise NotImplementedError(
+            f"Toga's Android backend doesn't implement '{name}'"
+        ) from None
+    else:
+        module = importlib.import_module(module_name)
+        value = getattr(module, name) if has_dot else module
```

Comment:
This logic works, but it's a bit convoluted - it involves evaluating `has_dot` twice. You can get the same effect in less lines of code using `rsplit()` and iteration:
```
        base_name, *extra = module_name.rsplit('.', 1)
        value = importlib.import_module(base_name)
        for name in extra:
            value = getattr(value, name)
```